src/components/form-elements/radio/Radio.jsx

```diff
@@ -0,0 +1,158 @@
+// @flow strict
+
+import * as React from 'react';
+import classNames from 'classnames';
+import Text from '../../text/Text';
+import generateRandomString from '../../../js/generateRandomString';
+import {useRadioContext} from './useRadioContext';
+
+export type RadioColorType = 'light' | 'dark';
+
+export type RadioPropsType = {
+  checked?: boolean,
+  children?: React.Node,
+  className?: ?string,
+  color?: ?RadioColorType,
+  description?: React.Node | string,
+  disabled?: boolean,
+  id?: string,
+  invalid?: boolean,
+  name?: string,
+  onChange: (SyntheticInputEvent<HTMLInputElement>) => mixed,
+  required?: boolean,
+  /**
+   * Style applied to the container.
+   * @example <Radio style={{ '--radioColor': '#000' }} />
+   */
+  style?: $Shape<CSSStyleDeclaration>,
+  value: string,
+  'aria-labelledby'?: string,
+  'aria-describedby'?: string,
+  ...
+};
+
+const Radio = ({
+  checked,
+  color = 'dark',
+  children,
+  className,
+  description,
+  disabled,
+  id,
+  invalid,
+  name,
+  onChange,
+  required = false,
+  style,
+  value,
+  'aria-labelledby': ariaLabelledBy,
+  'aria-describedby': ariaDescribedBy,
+  ...props
+}: RadioPropsType) => {
+  const {current: radioId} = React.useRef(
+    id === undefined || id === '' ? generateRandomString() : id
+  );
+
+  const radioGroupContext = useRadioContext();
+  const isWithinRadioGroup =
+    radioGroupContext && Object.keys(radioGroupContext).length;
+
+  const isDisabled =
+    disabled !== undefined ? disabled : radioGroupContext.disabled;
+  const isInvalid = invalid !== undefined ? invalid : radioGroupContext.invalid;
+  const isControlled = checked !== undefined || isWithinRadioGroup;
+  let isChecked = undefined;
+
+  if (isControlled) {
+    // Radio can either be directly set as checked, or be controlled by a RadioGroup
+    isChecked =
+      checked !== undefined
+        ? checked
+        : radioGroupContext.selectedValue &&
+          radioGroupContext.selectedValue === value;
+  }
+  const hasLabel = children !== undefined && children !== null;
+  const labelId = ariaLabelledBy || `${radioId}-label`;
+
+  const descriptionId = React.useMemo(() => {
+    if (ariaDescribedBy) return ariaDescribedBy;
+    if (description) return `${radioId}-description`;
+    return null;
+  }, [radioId, ariaDescribedBy, description]);
+
+  const colorName = radioGroupContext.color || color;
+  const radioClass = classNames('sg-radio', className, {
+    [`sg-radio--${String(colorName)}`]: colorName,
+    'sg-radio--disabled': isDisabled,
+    'sg-radio--with-label': !!hasLabel,
+    'sg-radio--with-description': !!descriptionId,
+  });
+  const labelClass = classNames('sg-radio__label', {
+    'sg-radio__label--with-padding-bottom': description,
```

Comment:
@clxandstuff it is actually done on purpose, it is not so easy as it seems 😅 Take a look at the layout. Both label and checkbox are in the same container, so we can handle hover states properly. Checkbox is 32x32 and because of that it moves the description down. In that case the spacing between label and description looks fine. If we would add padding-top to the description, it would make this spacing too big. So, that is why the description has this padding - if it will be multiline it will still preserve correct padding from the description. Let me know if that makes sense